docs/content/docs/guide/repositorycrd.md

```diff
@@ -78,3 +91,115 @@ request with a `concurrency_limit` of 1 in the repository configuration, then al
 of the pipelineruns will be executed in alphabetical order, one after the
 other. At any given time, only one pipeline run will be in the running state,
 while the rest will be queued.
+
+## Custom Parameter Expansion
+
+Using the `{{ param }}` syntax, Pipelines as Code let you expand a variable
+inside a template directly within your PipelineRuns.
+
+By default, there are
+several variables exposed according to the event. To view all the variables
+exposed by default, refer to the documentation on [Authoring
+PipelineRuns](../authoringprs).
+
+With the custom Parameter expansion, you can specify some custom values to be
+replaced inside the template.
+
+{{< hint warning >}}
+Utilizing the Tekton PipelineRun parameters feature may generally be the
+preferable approach, and custom params expansion should only be used in specific
+scenarios where Tekton params cannot be used.
+{{< /hint >}}
+
+As an example here is a custom variable in the Repository CR `spec`:
+
+```yaml
+spec:
+  params:
+    - name: company
+      value: "My Beautiful Company"
+```
+
+The variable name `{{ company }}` will be replaced by `My Beautiful Company`
+anywhere inside your `PipelineRun` (including the remotely fetched task).
+
+Alternatively, the value can be retrieved from a Kubernetes Secret.
+For instance, the following code will retrieve the value for the company
+`parameter` from a secret named `my-secret` and the key `companyname`:
+
+```yaml
+spec:
+  params:
+    - name: company
+      secretRef:
+        name: my-secret
+        key: companyname
+```
+
+{{< hint info >}}
+
+- If you have a `value` and a `secretRef` defined, the `value` will be used.
+- If you don't have a `name` in the `params` the parameter will not parsed.
+- If you have multiple `params` with the same `name` the last one will be used.
```

Comment:
what happens if name is given but no secret/value/filters

Reply:
same comment as before, it would just not fail and can be used as is.. maybe we should erroring but since it will have the same behaviour as when you don't have params i think it's safe?